File: ControlLib/src/main/java/org/marsroboticsassociation/controllib/hardware/IMotor.java
package org.marsroboticsassociation.controllib.hardware;

public interface IMotor {
    String getName();

    int getPosition(); // encoder ticks

    double getVelocity(); // ticks per second

    void setPower(double power);

    double getHubVoltage(); // volts, live read

    void setVelocity(double tps);

    void setVelocityPIDFCoefficients(double p, double i, double d, double f);
}

File: ControlLib/src/main/kotlin/org/marsroboticsassociation/controllib/hardware/IMotor.kt
package org.marsroboticsassociation.controllib.hardware

interface IMotor {
    val name: String

    /** Encoder ticks. */
    val position: Int

    /**
     * Encoder ticks per second. Named separately from `DcMotorEx.getVelocity()` so a Kotlin class
     * can implement both without a JVM signature clash (that Java method is overloaded).
     */
    val encoderVelocity: Double

    fun setPower(power: Double)

    /** Volts, live read. */
    val hubVoltage: Double

    fun setVelocity(tps: Double)

    fun setVelocityPIDFCoefficients(p: Double, i: Double, d: Double, f: Double)
}
